### PR TITLE
update to latest reference tests develop branch

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description
* update ref tests to latest develop
* add London to the list of forks tested
* handle missing baseFee for legacy tests
* ignore `transactionSequence` for blockchain test cases (per-transaction block inclusion/exclusion reason)
* remove aleut spec

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/GeneralStateReferenceTestTools.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ private static MainnetTransactionProcessor transactionProcessor(final String nam
     final String eips =
         System.getProperty(
             "test.ethereum.state.eips",
-            "Frontier,Homestead,EIP150,EIP158,Byzantium,Constantinople,ConstantinopleFix,Istanbul,Berlin");
+            "Frontier,Homestead,EIP150,EIP158,Byzantium,Constantinople,ConstantinopleFix,Istanbul,Berlin,London");
```
Comment (RatanRSur) [inline ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/GeneralStateReferenceTestTools.java:63]: Is there something we can do to make this automatic? At the very least, can we add Shanghai, Cancun, etc. here now so that we don't forget them?

Comment (garyschulte) [reply ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/GeneralStateReferenceTestTools.java:63]: will pre-add them.  trying to introspect into the test jsons to get the full set of forks is probably just too messy.

File: ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/GeneralStateTestCaseSpec.java [hunk 2]
```diff
@@ -120,6 +136,7 @@ public int hashCode() {
   }
 
   /** Represents the "post" part of a general state test json _for a specific hard-fork_. */
+  @JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (RatanRSur) [inline ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/GeneralStateTestCaseSpec.java:139]: Why is this needed?

**Review by RatanRSur — APPROVED** [review]
Just 1 more question, probably not a blocker.

**Comment by RatanRSur** [comment]
Oh, and can we also remove baikal from the list of state test forks?

File: ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/GeneralStateTestCaseSpec.java (hunk 2 above)
Comment (garyschulte) [reply ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/GeneralStateTestCaseSpec.java:139]: The latest test artifacts have transaction exception / failure expectations that we are not currently using.

**Comment by garyschulte** [comment]
> Oh, and can we also remove baikal from the list of state test forks?

The only reference I can find to baikal is in protocol schedules, and Karim has repurposed it for Calaveras.
